include/seqan3/core/tuple_utility.hpp

```diff
@@ -0,0 +1,184 @@
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides utility functions for tuple like interfaces.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <meta/meta.hpp>
+
+#include <utility>
+
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/core/type_list.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Helper function for seqan3::tuple_split.
+ * \ingroup core
+ *
+ * \tparam    beg A template value containing the start position from where to extract the values.
+ *
+ * \param[in] t   The original tuple to split.
+ * \param[in] idx An std::index_sequence with all indices that should be extracted beginning at `beg`.
+ *
+ * \returns A new tuple with the extracted elements.
+ */
+template <size_t beg,
+          template <typename ...> typename tuple_t,
+          size_t ... Is,
+          typename ...ts>
+          requires tuple_like_concept<tuple_t<ts...>> && tuple_like_concept<tuple_t<>>
+constexpr auto tuple_split(tuple_t<ts...> const & t, std::index_sequence<Is...> const & SEQAN3_DOXYGEN_ONLY(idx))
+{
+    return tuple_t<std::tuple_element_t<beg + Is, tuple_t<ts...>>...>{std::get<beg + Is>(t)...};
+}
+
+//!\copydoc seqan3::detail::tuple_split
+template <size_t beg,
+          template <typename ...> typename tuple_t,
+          size_t ... Is,
+          typename ...ts>
+          requires tuple_like_concept<tuple_t<ts...>> && tuple_like_concept<tuple_t<>>
+constexpr auto tuple_split(tuple_t<ts...> && t, std::index_sequence<Is...> const & SEQAN3_DOXYGEN_ONLY(idx))
+{
+    return tuple_t<std::tuple_element_t<beg + Is, tuple_t<ts...>>...>{std::move(std::get<beg + Is>(t))...};
+}
+}
+
+namespace seqan3
+{
+
+/*!\brief Splits a tuple like data structure at the given position.
+ * \ingroup core
+ *
+ * \tparam    pivot_c A template value specifying the split position.
+ * \param[in] t       The original tuple to split.
+ *
+ * \returns A new tuple of tuples with the left side of the split and the right side of the split.
+ *
+ * \details
+ *
+ * Splits a tuple into two tuples, while the element at the split position will be contained in the second tuple.
+ * Note, that the returned tuples can be empty. For this reason it is not possible to use tuple like objects,
+ * that cannot be empty, i.e. std::pair. Using such an object will emit an compiler error.
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements.
+ *
+ * ### Thread safety
+ *
+ * Concurrent invocations of this functions are thread safe.
+ */
+template <size_t pivot_c, template <typename ...> typename tuple_t, typename ... ts>
+    requires tuple_like_concept<tuple_t<ts...>>
+constexpr auto tuple_split(tuple_t<ts...> const & t)
+{
+    static_assert(pivot_c < sizeof...(ts));
```

Comment:
I would allow the corner case `pivot_c == sizeof...(ts)`, this makes subsequent code easier.